### PR TITLE
Auth: Cache Auth Info

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

Cache auth info to avoid multiple retrieval

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/infra/remotecache/redis_storage.go [hunk 1]
```diff
@@ -93,7 +94,15 @@ func (s *redisStorage) Set(ctx context.Context, key string, data []byte, expires
...
-	return s.c.Get(ctx, key).Bytes()
+	item, err := s.c.Get(ctx, key).Bytes()
+	if err != nil {
+		if errors.Is(err, redis.Nil) {
+			return nil, ErrCacheItemNotFound
+		}
+		return nil, err
+	}
+
+	return item, nil
```
Comment (kalleep) [inline pkg/infra/remotecache/redis_storage.go:105]: Nice catch

**Review by kalleep — APPROVED** [review]
LGTM

**Review by gamab — APPROVED** [review]
LGTM

File: pkg/infra/remotecache/redis_storage.go (hunk 1 above)
Comment (eleijonmarck) [reply pkg/infra/remotecache/redis_storage.go:105]: we should have enforced handled errors in our linter no?

Comment (kalleep) [reply pkg/infra/remotecache/redis_storage.go:105]: This just fix an inconsistency where all other backing storages for remote caching was returning `ErrCacheItemNotFound` and the _redis_ storage return `redis.Nil`. It still returned an error but a different one

**Pull request merged**